walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,79 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        final long meetingDuration = request.getDuration(); // in minutes
+        Collection<String> requestAttendees = request.getAttendees();
+
+        Collection<TimeRange> availableTimes = new ArrayList<>();
+
+        if (meetingDuration > TimeRange.WHOLE_DAY.duration()) {
+            return availableTimes;
+        }
+        if (events.size() == 0) {
+            availableTimes.add(TimeRange.WHOLE_DAY);
+            return availableTimes;
+        }
+
+        for (Event event : events) {
+            TimeRange when = event.getWhen();
+            Set<String> eventAttendees = event.getAttendees();
+
+            boolean hasNoCommonAttendees = Collections.disjoint(requestAttendees, eventAttendees);
+             
+            // if no event attendees need to go to requested meeting
+            if (hasNoCommonAttendees) {
+                if (availableTimes.size() == 0) {
+                    availableTimes.add(TimeRange.WHOLE_DAY);
+                }
+                continue;
+            }
+
+            // creates TimeRange available before and after event
+            TimeRange availableBefore = TimeRange.fromStartEnd(TimeRange.START_OF_DAY, when.start(), false);
```

Comment:
whenever you're passing in non-variable values like "false" or "true", you can make it more readable by adding the variable name in a comment. That way the reader doesn't need to reference the API in order to know what the code is doing. 

So here, it would be: 
`(....., when.start(), /* inclusive= */ false);`
`(....., TimeRange.END_OF_DAY, /* inclusive= */ true);`